util/compression/compression.go

```diff
@@ -14,24 +17,42 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type Compressor func(dest io.Writer, mediaType string) (io.WriteCloser, error)
+type Decompressor func(ctx context.Context, cs content.Store, desc ocispecs.Descriptor) (io.ReadCloser, error)
+type Finalizer func(context.Context, content.Store) (map[string]string, error)
+
+type Compression interface {
+	Compress(comp Config) (compressorFunc Compressor, finalize Finalizer)
+	Decompress(ctx context.Context, cs content.Store, desc ocispecs.Descriptor) (io.ReadCloser, error)
+	NeedsConversion(ctx context.Context, cs content.Store, desc ocispecs.Descriptor) (bool, error)
+	NeedsComputeDiffBySelf() bool
+	OnlySupportOCITypes() bool
+	MediaType() string
+	String() string
+}
+
 // Type represents compression type for blob data.
-type Type int
+type Type Compression
+
+type (
+	uncompressedType int
+	gzipType         int
+	estargzType      int
+	zstdType         int
+)
```

Comment:
Not sure about this.
These should be probably a single type.

Reply:
thanks a lot for review, Akihiro!

Reply:
We need to implement the `Compression` interface for each compression type, so defined different types here.

Reply:
> We need to implement the `Compression` interface for each compression type, so defined different types here.

Would it be possible to define structs rather than types?

Reply:
Does it mean like?

```
type uncompressedType struct {}
```

It's should be ok.

Reply:
Fixed by changing to `type uncompressedType struct {}`.